### PR TITLE
Docs Revamp: Models documentation

### PR DESCRIPTION
**Pull request by NimaBoscarino**
This PR migrates content (and introduces some new content) to form a "Models" section, with several subsections. The scope of this section is to help readers understand how to find, use, upload, and manage models on the Hub.

Anything that's _specifically_ about using `huggingface_hub` is omitted from this section. I'm sure there are other ways to structure the hierarchy of the content in this section, so let me know if you have any ideas!

Note: Images have been uploaded to the repo here, but will be moved to the documentation-images dataset repo in a future PR (I'll do all the images all at once). Also, most headings have not been renamed. If we want to go with simpler headings (i.e. from this convo https://github.com/huggingface/hub-docs/issues/62#issuecomment-1103614077) then I figured that could happen in a separate PR, where all the headings in the docs get redone at the same time to ensure consistency.

### CODE REVIEW

File: docs/hub/models-the-hub.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+title: The Model Hub
+---
+
+<h1>The Model Hub</h1>
+
+## What is the Model Hub?
+
+The Model Hub is where the members of the Hugging Face community can host all of their model checkpoints for simple storage, discovery, and sharing. Pre-trained models are available to be downloaded using the [`huggingface_hub` client library](https://huggingface.co/docs/huggingface_hub/index) or with [`🤗 transformers`](https://huggingface.co/docs/transformers/index) for fine-tuning and other usages. You can even leverage the [Inference API](./models-inference) to use models in production.
```
Comment (stevhliu) [inline docs/hub/models-the-hub.md:9]:
```suggestion
The Model Hub is where the members of the Hugging Face community can host all of their model checkpoints for simple storage, discovery, and sharing. Download pre-trained models with the [`huggingface_hub` client library](https://huggingface.co/docs/huggingface_hub/index) or with 🤗 [`Transformers`](https://huggingface.co/docs/transformers/index) for fine-tuning and other usages. You can even leverage the [Inference API](./models-inference) to use models in production.
```

File: docs/hub/models-widgets.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+* [Named Entity Recognition](https://huggingface.co/spacy/en_core_web_sm?text=My+name+is+Sarah+and+I+live+in+London) using [spaCy](https://spacy.io/).
+* [Image Classification](https://huggingface.co/google/vit-base-patch16-224) using [🤗 Transformers](https://github.com/huggingface/transformers)
+* [Text to Speech](https://huggingface.co/julien-c/ljspeech_tts_train_tacotron2_raw_phn_tacotron_g2p_en_no_space_train) using [ESPnet](https://github.com/espnet/espnet).
+* [Sentence Similarity](https://huggingface.co/osanseviero/full-sentence-distillroberta3) using [Sentence Transformers](https://github.com/UKPLab/sentence-transformers).
+
+You can try out all the widgets [here](https://huggingface-widgets.netlify.app/).
+
+## Creating a Widget
+
+A widget is automatically created for your model when you upload it to the Hub. To determine which pipeline and widget to display (`text-classification`, `token-classification`, `translation`, etc.), we analyze information in the repo, such as the metadata provided in the model card and configuration files. This information is mapped to a single `pipeline_tag`. We choose to expose **only one** widget per model for simplicity.
```
Comment (stevhliu) [inline docs/hub/models-widgets.md:21]: Maybe link to the model-cards page so users can see where this is all controlled?

File: docs/hub/models-widgets.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+For most use cases, we determine the model type from the tags. For example, if there is `tag: text-classification` in the metadata, the inferred `pipeline_tag` will be `text-classification`.
+
+However, for `🤗 Transformers`, the model type is determined automatically from `config.json`. The architecture can determine the type: for example, `AutoModelForTokenClassification` corresponds to `token-classification`. If you're interested in this, you can see pseudo-code in [this gist](https://gist.github.com/julien-c/857ba86a6c6a895ecd90e7f7cab48046).
```
Comment (stevhliu) [inline docs/hub/models-widgets.md:25]:
```suggestion
However, for 🤗  Transformers, the model type is determined automatically from `config.json`. The architecture can determine the type: for example, `AutoModelForTokenClassification` corresponds to `token-classification`. If you're interested in this, you can see pseudo-code in [this gist](https://gist.github.com/julien-c/857ba86a6c6a895ecd90e7f7cab48046).
```

File: docs/hub/models-interacting.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+```bash
+git lfs install
+git clone <MODEL URL>
+```
+
+For detailed information on accessing the model, you can click on the "Use in Transformer" button on any model page.
+
+![Models can be used locally through the "Use in Transformer" button](../assets/hub/models-usage.png)
+
+If the model is compatible with `🤗 transformers`, you'll even receive snippets to help you get started.
```
Comment (stevhliu) [inline docs/hub/models-interacting.md:20]:
```suggestion
If the model is compatible with 🤗  Transformers, you'll even receive snippets to help you get started.
```

**Review by stevhliu — COMMENTED** [review]
Did a first pass and it looks great! I'll come back to this in a little bit once I've had some time to reflect and process everything :)

**Comment by julien-c** [comment]
for the assets (screenshots), if possible upload some higher-resolution ones. Here they're a bit blurry on retina screens

cc @gary149

**Comment by julien-c** [comment]
(you can pass them through tinypng.com before adding them to the repo if the files are too heavy

File: docs/hub/models-main.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+The Hugging Face Hub hosts many models for a [variety of machine learning tasks](https://huggingface.co/tasks). Models are stored in repositories, so they benefit from [all the features](./repositories-main) possessed by every repo on the Hugging Face Hub. Additionally, model repos have attributes that make exploring and using models as easy as possible. These docs will take you through everything you'll need to know to find models on the Hub, upload your models, and make the most of everything the Model Hub offers!
+
+## Contents
+
+- [The Model Hub](./models-the-hub)
+- [Model Cards](./models-cards)
+  - [CO<sub>2</sub> emissions](./models-cards-co2)
+- [Tasks](./models-tasks)
+- [Interacting with models on the Hub](./models-interacting)
```
Comment (julien-c) [inline docs/hub/models-main.md:15]: `Interacting with models` sounds vague to me, WDYT of changing to something like `downloading and uploading models to/from the hub`?

or even split into `download` and `upload`

Comment (NimaBoscarino) [reply docs/hub/models-main.md:15]: Sounds good to me! I put that in my to-do list for the final polishing pass :)

**Pull request merged**